packages/opencensus-web-instrumentation-zone/test/test-perf-resource-timing-selector.ts

```diff
@@ -0,0 +1,216 @@
+      // endTime.
+      expect(filteredPerfEntries).not.toContain([entry1, entry2]);
+      expect(filteredPerfEntries).toContain([entry1, entry3]);
+      expect(filteredPerfEntries).toContain([entry1, entry4]);
+      expect(filteredPerfEntries).not.toContain([entry2, entry3]);
+      expect(filteredPerfEntries).toContain([entry2, entry4]);
+      expect(filteredPerfEntries).not.toContain([entry3, entry4]);
+    });
+
+    it('Should take single resource timing entries', () => {
+      const entry1 = createFakePerfResourceEntry(
+        2.100000011036173,
+        5.500000011036173
+      );
+      const entry2 = createFakePerfResourceEntry(
+        5.500000011036173,
+        10.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        1.500000011036173,
+        20.000000000000003
+      );
+      const perfResourceEntries = [entry1, entry2, entry3];
+      const filteredPerfEntries = getPossiblePerfResourceEntries(
+        perfResourceEntries
+      );
+      expect(filteredPerfEntries.length).toBe(3);
+      expect(filteredPerfEntries).toContain(entry1);
+      expect(filteredPerfEntries).toContain(entry2);
+      expect(filteredPerfEntries).toContain(entry3);
+    });
+
+    it('Should take single and tuples of resource timing entries', () => {
+      const entry1 = createFakePerfResourceEntry(
+        2.100000011036173,
+        5.500000011036173
+      );
+      const entry2 = createFakePerfResourceEntry(
+        6.500000011036173,
+        10.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        8.500000011036173,
+        20.000000000000003
+      );
+      const entry4 = createFakePerfResourceEntry(
+        1.500000011036173,
+        25.000000000000003
+      );
+      const perfResourceEntries = [entry1, entry2, entry3, entry4];
+      const filteredPerfEntries = getPossiblePerfResourceEntries(
+        perfResourceEntries
+      );
+      expect(filteredPerfEntries.length).toBe(3);
+      expect(filteredPerfEntries).toContain([entry1, entry2]);
+      expect(filteredPerfEntries).toContain([entry1, entry3]);
+      expect(filteredPerfEntries).not.toContain([entry1, entry4]);
+      expect(filteredPerfEntries).not.toContain([entry2, entry3]);
+      expect(filteredPerfEntries).not.toContain([entry2, entry4]);
+      // Should not contain the single resource entries as they already were
+      // paired.
+      expect(filteredPerfEntries).not.toContain(entry1);
+      expect(filteredPerfEntries).not.toContain(entry2);
+      expect(filteredPerfEntries).not.toContain(entry3);
+      // As entry4 has not been paired, should contain this a single entry.
+      expect(filteredPerfEntries).toContain(entry4);
+    });
+  });
+
+  describe('getXhrPerfomanceData', () => {
+    it('Should take the tuple with the minimum gap to the span as the best resource timing entry', () => {
+      const entry1 = createFakePerfResourceEntry(
+        13.100000011036173,
```

Comment:
Optional/nit: what would you think about simplifying the numbers a bit here to make the gap differences more obvious?